### PR TITLE
fix(apigateway): move resource api_gateway_api into services and fix codecheck errors

### PR DESCRIPTION
**Pull request by wangyuancheng01**
**What this PR does / why we need it**: Mobile code path

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes # codecheck errors

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/apigateway" TESTARGS="-run TestAccApiGatewayAPI_basic"
...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/apigateway -v -run TestAccApiGatewayAPI_basic -timeout 360m -parallel 4
=== RUN   TestAccApiGatewayAPI_basic
=== PAUSE TestAccApiGatewayAPI_basic
=== CONT  TestAccApiGatewayAPI_basic
--- PASS: TestAccApiGatewayAPI_basic (28.28s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/apigateway        28.318s
```

### CODE REVIEW

File: huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go [hunk 1]
```diff
@@ -340,7 +348,7 @@ func resourceAPIGatewayAPIRead(d *schema.ResourceData, meta interface{}) error {
 		requestParameters = append(requestParameters, parameters)
 	}
 	if err = d.Set("request_parameter", requestParameters); err != nil {
-		return fmtp.Errorf("Saving request parameters failed: %s", err)
+		return diag.Errorf("saving request parameters failed: %s", err)
 	}
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go]:
```
mErr = multierror.Append(mErr, d.Set("request_parameter", requestParameters))
```

File: huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go [hunk 2]
```diff
@@ -390,43 +398,45 @@ func resourceAPIGatewayAPIRead(d *schema.ResourceData, meta interface{}) error {
...
+	apigwClient, err := cfg.ApiGatewayV1Client(region)
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud api gateway client: %s", err)
+		return diag.Errorf("error creating api gateway client: %s", err)
 	}
 
 	updateOpts, err := buildApiParameter(d)
 	if err != nil {
-		return err
+		return diag.Errorf("error creating api gateway parameter: %s", err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go]:
```suggestion
		return diag.Errorf("error creating api gateway options: %s", err)
```

File: huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go [hunk 3]
```diff
@@ -390,43 +398,45 @@ func resourceAPIGatewayAPIRead(d *schema.ResourceData, meta interface{}) error {
...
-		return fmtp.Errorf("Error creating HuaweiCloud api gateway client: %s", err)
+		return diag.Errorf("error creating api gateway client: %s", err)
 	}
 
 	if err := apis.Delete(apigwClient, d.Id()).ExtractErr(); err != nil {
-		return CheckDeleted(d, err, "api apis")
+		return common.CheckDeletedDiag(d, err, "api apis")
 	}
 
 	d.SetId("")
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go]:
```suggestion
```

File: huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go [hunk 4]
```diff
@@ -458,19 +468,19 @@ func buildApiParameter(d *schema.ResourceData) (*apis.CreateOpts, error) {
 	case "HTTP":
 		httpBackend := buildHttpBackendParam(d)
 		if httpBackend == nil {
-			return nil, fmtp.Errorf("The argument \"http_backend\" is required under HTTP backend type")
+			return nil, fmt.Errorf("The argument \"http_backend\" is required under HTTP backend type")
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go]:
```suggestion
			return nil, fmt.Errorf("the argument \"http_backend\" is required under HTTP backend type")
```

File: huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go (hunk 1 above)
Comment (wangyuancheng01) [reply huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go]: down

File: huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go (hunk 2 above)
Comment (wangyuancheng01) [reply huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go]: down

File: huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go (hunk 3 above)
Comment (wangyuancheng01) [reply huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go]: down

File: huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go (hunk 4 above)
Comment (wangyuancheng01) [reply huaweicloud/services/apigateway/resource_huaweicloud_api_gateway_api.go]: down

**Pull request merged**